lib/sqlalchemy/orm/strategies.py

```diff
@@ -251,15 +251,24 @@ def setup_query(
         if loadopt and "expression" in loadopt.local_opts:
             columns = [loadopt.local_opts["expression"]]
 
-            for c in columns:
-                if adapter:
-                    c = adapter.columns[c]
-                column_collection.append(c)
+        else:
+            columns = self.parent_property.columns
+
+        columns_are_null = all([
+            c.expression.type._isnull for c in columns
```

Comment:
yeah this part will do it.    I might alter this approach however to be declared up at the top when the object is first created, so that it is only called once, as this setup_query() method is called for every query

Reply:
This condition test is moved to __init__. It should be executed only once for each query_expression now :)